src/app/components/client/Button.tsx

```diff
@@ -35,6 +35,10 @@ export interface Props {
   theme?: ButtonColorThemes;
 }
 
+type Props =
+  | (BaseProps & { variant: "link"; small?: never })
+  | (BaseProps & { variant: Exclude<ButtonVariants, "link">; small?: boolean });
```

Comment:
Buttons of variant `link` are only supposed to be used inline and don’t have a small alternate.